packages/frontend/src/components/TestSubstep/index.tsx

```diff
@@ -142,7 +143,9 @@ function TestSubstep(props: TestSubstepProps): React.ReactElement {
               </AlertTitle>
 
               <Box sx={{ fontWeight: 400 }}>
-                {formatMessage('flowEditor.noTestDataMessage')}
+                {(selectedActionOrTrigger as IBaseTrigger)
```

Comment:
`hasNoOutput` can be true when `selectedActionOrTrigger` is an `IAction`, so we shouldn't cast to `IBaseTrigger` here.

Reply:
That said... your code still works. But it's not ideal because its communicating the wrong types.

For fun, here's a convoluted one liner to do that in a type-safe manner using TS narrowing:

```
{selectedActionOrTrigger &&
  'webhookTriggerInstructions' in selectedActionOrTrigger &&
  selectedActionOrTrigger.webhookTriggerInstructions?.errorMsg
    ? selectedActionOrTrigger.webhookTriggerInstructions.errorMsg
    : formatMessage('flowEditor.noTestDataMessage')}
```

Reply:
ok thanks a lot!